src/pyfmi/fmi3.pyx

```diff
@@ -675,6 +678,155 @@ cdef class FMUModelBase3(FMI_BASE.ModelBase):
             raise FMUException(f"The variable {pyfmi_util.decode(variablename)} could not be found.")
         return FMIL3.fmi3_import_get_variable_base_type(variable)
 
+
+    #@enable_caching TODO: Doesnt work using enums with members with C-types, lets discuss
+    def get_model_variables(self,
+        type: Optional[FMI3_Type] = None,
+        include_alias: bool = True,
+        causality: Optional[FMI3_Causality] = None,
+        variability: Optional[FMI3_Variability] = None,
+        only_start: bool = False,
+        only_fixed: bool = False,
+        filter = None,
+        _as_list: bool = False) -> Dict[str, FMI3ModelVariable]:
+        """
+        Extract the names of the variables in a model.
+
+        Parameters::
+
+            type --
+                The type of the variables as an instance of pyfmi.fmi3.FMI3_Type.
+                Default: None (i.e all).
+
+            include_alias --
+                Currently not supported (does nothing).
+                If alias should be included or not.
+                Default: True
+
+            causality --
+                The causality of the variables as an instance of pyfmi.fmi3.FMI3_Causality.
+                Default: None (i.e all).
+
+            variability --
+                The variability of the variables as an instance of pyfmi.fmi3.FMI3_Variability.
+                Default: None (i.e all).
+
+            only_start --
+                If only variables that has a start value should be returned.
+                Default: False
+
+            only_fixed --
+                If only variables that has a start value that is fixed should be returned.
+                Default: False
+
+            filter --
+                Filter the variables using a unix filename pattern
+                matching (filter="*der*"). Can also be a list of filters
+                See http://docs.python.org/2/library/fnmatch.html.
+                Default: None
+
+        Returns::
+
+            Dict with variable name as key and a pyfmi.fmi3.FMI3ModelVariable class as value.
+        """
+        cdef FMIL3.fmi3_import_variable_t*        variable
+        cdef FMIL3.fmi3_import_variable_list_t*   variable_list
+        cdef FMIL.size_t                          variable_list_size
+        cdef FMIL3.fmi3_value_reference_t         value_ref
+        cdef FMIL3.fmi3_base_type_enu_t           data_type
+        cdef FMIL3.fmi3_variability_enu_t         data_variability
+        cdef FMIL3.fmi3_causality_enu_t           data_causality
+        cdef FMIL3.fmi3_initial_enu_t             data_initial
+
+        cdef int has_start = 0
+        # TODO: Can we rename the keyword arg 'filter' to variable_filter?
```

Comment:
As much as I dislike the use of keywords as as parameter names here, we may want to keep it for consistency and revisit for a 3.x release, when we can break things.

Reply:
I agree.
I thought also we could do this:
current:
```
def foo(self, ..., type = None, ...)
```
new:
```
def foo(self, ..., variable_type = None, ..., type = None):
if type is not None:
  # inform user that 'type' is deprecated and they should use variable_type
  # check if user specified both  and raise error (no user will a be affected by this yet, but could in the future)
   variable_type = type
```
This way, the order of the argument has not changed, and if the user has specified `type=...` to something, then they will see the warning. Thoughts? I'm also fine to just leave it for now but wanted to share my idea.

Reply:
Not sure if there is much point in introducing this as deprecated for FMI3 to begin with. As I see it, we want this to be gone, yet be consistent with FMI1 & 2, so these are the problematic parts.

I think this would be a great solution if we wanted to move away from this as a whole, including FMI1 & 2.